internal/pkg/docker/dockercompose/decompose.go

```diff
@@ -126,3 +137,38 @@ func unsupportedServiceKeys(service map[string]any, svcName string) (IgnoredKeys
 
 	return ignored, nil
 }
+
+// isolateEnvFiles manually handles the env_file key so that compose-go does not try to read the env files from the disk.
+func isolateEnvFiles(service map[string]any) ([]string, error) {
+	envFile, ok := service["env_file"]
+	if !ok || envFile == nil {
+		return nil, nil
+	}
+
+	if envFileStr, ok := envFile.(string); ok {
+		return []string{envFileStr}, nil
```

Comment:
not really a suggestion, bc it's probably better the way you have it bc of the nested-ness, but wanted to tell you about the type switch
```
switch v := envFile.(type) {
case string:
  // can use v as a string
case []any:
default:
}
```

Reply:
I didn't consider that feature, thanks for reminding me! I think you're probably right though about the nested-ness, though, that is one thing I was aiming to avoid here 👍